python/ucxx/ucxx/_lib/libucxx.pxd

```diff
@@ -0,0 +1,103 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: BSD-3-Clause
+
+
+from libc.stdint cimport uint64_t, uintptr_t
+from libcpp.memory cimport shared_ptr, unique_ptr
+
+from .ucxx_api cimport *
+
+
+cdef class HostBufferAdapter:
+    cdef:
+        Py_ssize_t _size
+        void* _ptr
+        Py_ssize_t[1] _shape
+        Py_ssize_t[1] _strides
+        Py_ssize_t _itemsize
+
+    @staticmethod
+    cdef _from_host_buffer(HostBuffer* host_buffer)
+
+
+cdef class UCXConfig:
+    cdef:
+        unique_ptr[Config] _config
+        bint _enable_python_future
+        dict _cb_data
+
+
+cdef class UCXContext:
+    cdef:
+        shared_ptr[Context] _context
+        dict _config
+
+    cpdef dict get_config(self)
+
+    cdef shared_ptr[Context] get_ucxx_shared_ptr(self) nogil
+
+
+cdef class UCXAddress:
+    cdef:
+        shared_ptr[Address] _address
+        size_t _length
+        ucp_address_t *_handle
+        string _string
+
+    cdef shared_ptr[Address] get_ucxx_shared_ptr(self) nogil
```

Comment:
suggestion/question: What does this `get_ucxx_shared_ptr` setup buy us?

Is it reasonable to instead write:
```
cdef class UCXFoo:
    cdef:
        shared_ptr[Foo] c_obj
```
And just access `c_obj` directly?

Reply:
Ok, makes sense.